scripts/emoji_codegen.py

```diff
@@ -68,16 +74,65 @@ def generate_qml_list(**kwargs):
         code, qualification, charAndName = segments
 
         # skip unqualified versions of same unicode
-        if qualification == 'unqualified':
-            continue
-
-        if qualification == 'component':
+        if qualification != 'fully-qualified':
             continue
+        
 
         char, name = re.match(r'^(\S+) E\d+\.\d+ (.*)$', charAndName).groups()
-
-        categories[current_category].append(Emoji(code, name))
+        shortname = name
+        # until skin tone is handled, keep them around
+        # discard skin tone variants for sanity
+        # __contains__ is so stupid i hate prototype languages
```

Comment:
You wouldn't call `__contains__` directly, you would do `if "skin tone" in name:`